docs/guide/db-active-record.md

```diff
@@ -1422,3 +1422,84 @@ $customers = Customer::find()
     ->groupBy('{{customer}}.id') // group the result to ensure aggregation function works
     ->all();
 ```
+
+A disadvantage of using this method would be that if the information isn't loaded on the sql query it has to be calculated separatedly, which also means that newly saved records won't contain the information from any extra field
+
+```php
+$room = new Room();
+$room->length = 100;
+$room->width = 50;
+$room->height = 2;
+
+$room->volume; // this value will be null since it was not declared yet.
+```
+
+Using the magic methods [[yii\db\BaseActiveRecord::__get()|__get()]] and [[yii\db\BaseActiveRecord::__set()|__set()]] we can emulate the behavior of a property
+
+```php
+class Room extends \yii\db\ActiveRecord
+{
+    private $_volume;
+    
+    public function setVolumen($volume)
+    {
+        $this->_volume = (float) $volume;
+    }
+    
+    public function getVolume()
+    {
+        if (empty($this->length)
+            or empty($this->width)
+            or empty($this->height)
+        ) {
+            return null;
+        }
+        
+        if ($this->_volume === null) {
+            $this->setVolume(
+                $this->length * $this->width * $this->height
+            );
+        }
+        
+        return $this->_volume;
+    }
+
+    // ...
+}
+```
+
+With this if the select query doesn't provide the volume the record will obtain it from its internal value.
```

Comment:
It will be automatic, not manually since the user won't need to declare it. I think we can change it to.

> When the select query doesn't provide the volume, model will be able to calculate it automatically from the attributes in the model.